packages/wallet-management/src/walletAutomation.ts

```diff
@@ -1,67 +1,52 @@
 /* eslint-disable radix */
+import { Web3Provider } from '@ethersproject/providers';
 import type { StaticToken } from '@lifi/sdk';
 import {
   MetaMaskProviderErrorCode,
   getChainById,
   prefixChainId,
 } from '@lifi/sdk';
-import type { Provider } from './types';
 
 export const switchChain = async (
-  provider: Provider,
+  provider: Web3Provider,
   chainId: number,
 ): Promise<boolean> => {
-  return new Promise((resolve, reject) => {
-    try {
-      provider
-        .request({
-          method: 'wallet_switchEthereumChain',
-          params: [{ chainId: getChainById(chainId).metamask?.chainId }],
-        })
-        .catch((error: any) => {
-          if (error.code !== MetaMaskProviderErrorCode.userRejectedRequest) {
-            addChain(provider, chainId).then((result) => resolve(result));
-          } else {
-            reject(error);
-          }
-        });
-      provider.once('chainChanged', (id: string) => {
-        if (parseInt(id) === chainId) {
-          resolve(true);
-        }
-      });
```

Comment:
The last time we talked about this, we decided to keep it. Could you please share why you removed it? 🙂

Reply:
The .send() works differently. By testing out the new flow I noticed that this was not necessary anymore. Switching chains works without it